### DISCUSSION TITLE
Relaying messages (node mesh/routing) questions

### DISCUSSION THREAD

**Discussion by Nume1977** [gioblu/PJON#391] [Q&A] [answered]
Hello all,
I have been exploring Pjon mainly via RS485 and everything is quite straight forward and works well. (Still noob, so don't flame please ;) )

Now I would like to turn my attention to LoRa radios using individual 'mac' addresses for the purpose of transmitting sensor data from a few hundred (over 255) nodes and only have 4/5 end receivers spread over a 20km city area.

After reading every single example for Pjon I did not see any example where a node could receive and retransmit a particular message that is addressed to another node, thus creating a mesh (important for unwired communication, since distance could be a problem).

The closest thing i read was [Routing](https://www.pjon.org/routing.php), but i don't think it is the same thing.

So my questions are:

Does each node retransmit the data until it reaches the destination receiver when using radios (LoRa)?
Is there any example i can study? (google didn't help) ...

Thanks in advance :)

Rui

**Comment by gioblu** [comment, marked as answer]
Ciao @Nume1977, if you want to use the MAC feature you must define `PJON_INCLUDE_MAC` before including PJON, then you should call `set_router(true)` (which forces the device to receive all packets, also if not addressed to itself) and use the `forward` function to forward a packet to another device when that is received. If you use PJON in shared mode the hop count is included in the packet's meta-data, so when the maximum hop count is reached the packet is discarded (avoiding infinite loops in the network).  Then it is your choice if to keep an array of known devices in each device or if using just flooding.

Here you find a MAC example: 
https://github.com/gioblu/PJON/tree/master/examples/ARDUINO/Local/SoftwareBitBang/BlinkTestMAC
See also the surrogate example: 
https://github.com/gioblu/PJON/blob/9e6345588f3273f62d1a3d2b36e896b12285bf8a/examples/ARDUINO/Local/ThroughSerial/SoftwareBitBangSurrogate/Surrogate/Surrogate.ino
And the SWBB-MQTT gateway:
https://github.com/gioblu/PJON/blob/9e6345588f3273f62d1a3d2b36e896b12285bf8a/examples/ARDUINO/Local/MQTTTranslate/SWBB-MQTT-Gateway/Gateway/Gateway.ino

**Reply by Nume1977** [reply]
Grazie mille @gioblu !

I am now starting to understand how modular and robust Pjon really is, I have only used it for basic communication before.
My mistake was I was looking at each example individually, when I should have been looking at the big picture :)

**Comment by gioblu** [comment]
@Nume1977 please let me know if you need any help.

**Reply by Nume1977** [reply]
@gioblu thank you for your assistance, if I get stuck I will contact you for sure.

I will start with a basic test setup over the next few days, if I manage to make a good test run, I will publish an example for others to read :)

**Reply by gioblu** [reply]
Ciao @Nume1977 yes, that would be really helpful, please open a pull-request when you have it working, I bet a lot of people will find it useful.

**Reply by gioblu** [reply]
@Nume1977 if you need direct support you find us here: https://gitter.im/gioblu/PJON
